holoviews/operation/downsampling.py

```diff
@@ -0,0 +1,176 @@
+"""
+Implements downsampling algorithms for large 1D datasets.
+
+The algorithms implemented in this module have been adapted from
+https://github.com/predict-idlab/plotly-resampler and are reproduced
+along with the original license:
+
+MIT License
+
+Copyright (c) 2022 Jonas Van Der Donckt, Jeroen Van Der Donckt, Emiel Deprost.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import math
+
+import numpy as np
+import param
+
+from .resampling import ResamplingOperation1D
+
+
+def _argmax_area(prev_x, prev_y, avg_next_x, avg_next_y, x_bucket, y_bucket):
+    """Vectorized triangular area argmax computation.
+    Parameters
+    ----------
+    prev_x : float
+        The previous selected point is x value.
+    prev_y : float
+        The previous selected point its y value.
+    avg_next_x : float
+        The x mean of the next bucket
+    avg_next_y : float
+        The y mean of the next bucket
+    x_bucket : np.ndarray
+        All x values in the bucket
+    y_bucket : np.ndarray
+        All y values in the bucket
+    Returns
+    -------
+    int
+        The index of the point with the largest triangular area.
+    """
+    return np.abs(
+        x_bucket * (prev_y - avg_next_y)
+        + y_bucket * (avg_next_x - prev_x)
+        + (prev_x * avg_next_y - avg_next_x * prev_y)
+    ).argmax().astype(np.uint64)
```

Comment:
This is needed because `np.int64(0) + np.uint64(0)`  will return a float

Reply:
I have gone back to `int64`. The original idea was to support as large an array as possible. But there is a hard limit before that, e.g., this `np.empty(np.iinfo(np.int64).max)` will raise a `ValueError` about the size being too big.